selfdrive/car/gm/carstate.py

```diff
@@ -150,7 +150,7 @@ def update(self, pt_cp, cam_cp, loopback_cp):
       else:
         ret.stockAeb = False
       # openpilot controls nonAdaptive when not pcmCruise
-      if self.CP.pcmCruise:
+      if self.CP.pcmCruise and self.CP.carFingerprint not in CC_ONLY_CAR:
```

Comment:
I'd split this into a nested `if` block. From memory, I think that the value is always 4 for CC cars when cruise main is on, 1 otherwise. Can you check this?

Reply:
CC only cars are always going to be in "nonAdaptive" mode, so whats the point of checking since "ret.cruiseState.enabled" will control engagement now?

Reply:
I think the value can go to 1 for other reasons, such as traction control. But you're right, it's probably not a big deal.